### PR TITLE
Suggestion: use keyword.is<T>() instead of checking name?

### PR DESCRIPTION
**Pull request by joakim-hove**
I am in the process of cleaning up parts of the deck API. The codebase has literally hundreds of calls like:

```C++
if (keyword.name() == "XYZ") {  
    ....
}
```
in this PR I suggest replacing that with (only a couple of callsites to demonstrate):
```C++
if (keyword.is<ParserKeywords::XYZ>()) {
    ....
}
```

good idea?

### CODE REVIEW

File: opm/parser/eclipse/Deck/DeckKeyword.hpp [hunk 1]
```diff
@@ -77,8 +77,8 @@ namespace Opm {
         void write_TITLE( DeckOutput& output ) const;
 
         template <class Keyword>
-        bool isKeyword() const {
-            if (Keyword::keywordName == m_keywordName)
+        bool is() const {
+            if (Keyword::keywordName == this->m_keywordName)
                 return true;
             else
                 return false;
```
Comment (blattms) [inline opm/parser/eclipse/Deck/DeckKeyword.hpp:84]: This is strange code. Why not just:
```
return Keyword::keywordName == this->m_keywordName;
```

**Review by blattms — COMMENTED** [review]
The PR does nothing but renaming the function and how it is called.

If we will still compare strings then the introduction of  a template parameter  just brings up compile time and reduces readability.

Always wondering why we at OPM are so found of string comparisons.

Isn't  there a class for every keyword or an `enum` or `integral_constant` that one can compare with a simple constexpr function like `is_same`?

Of course I know nothing about this part of the code...

File: opm/parser/eclipse/Deck/DeckKeyword.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/parser/eclipse/Deck/DeckKeyword.hpp:84]: No reason - I'll change it.

**Comment by joakim-hove** [comment]
Have updated all the callsites I found at a first glance - were not as many as I had thought.

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
I think this is an improvement over the current state.  You may want to include the [`RPTONLY`/`ROTONLYO` handling](https://github.com/OPM/opm-common/blob/9d1f512b44755c88749a910d5b6a2c161860b898/src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp#L116-L119) too, but otherwise I'd be happy to merge this.

**Comment by joakim-hove** [comment]
> You may want to include the RPTONLY/ROTONLYO handling

Sure - that was an oversight.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**